Refactored WiremockCustomizer to take a CustomizationContext as argument

SourcePond opened the pull request:
- Introduced a parameter class CustomizationContext which additionally provides ExtensionContext and ParameterContext objects. This allows to implement more sophisticated setup logic based on the execution context.
- Added WiremockCustomizer::customize(WireMockServer server, CustomizationContext ctx) as default method to keep backward compatibility.
- Made WiremockCustomizer::customize(WireMockServer server) also a default method. This is to make sure, that only the necessary "customize" method needs be implemented on a WiremockCustomizer implementation (see TicketEndpoint).
- Allow to throw an exception from WiremockCustomizer::customize
- Added Unit-Tests
- Updated dependencies and plugin versions where possible
- Bumped version to 2.0.0-SNAPSHOT because the interface change

SourcePond committed:
Refactored WiremockCustomizer to take a CustomizationContext as argument
- Introduced a parameter class CustomizationContext which additionally provides ExtensionContext and ParameterContext objects. This allows to implement more sophisticated setup logic based on the execution context.
- Added WiremockCustomizer::customize(WireMockServer server, CustomizationContext ctx) as default method to keep backward compatibility.
- Made WiremockCustomizer::customize(WireMockServer server) also a default method. This is to make sure, that only the necessary "customize" method needs be implemented on a WiremockCustomizer implementation (see TicketEndpoint).
- Allow to throw an exception from WiremockCustomizer::customize
- Added Unit-Tests
- Updated dependencies and plugin versions where possible
- Bumped version to 2.0.0-SNAPSHOT because the interface change

SourcePond committed:
Fixed assertion

SourcePond committed:
Use NoopWiremockCustomizer to test default methods (code coverage)

SourcePond committed:
Pass correct replacement to exception message

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/CustomizationContextTest.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Test
+    public void verifyToString() {
+        System.out.println(customizationContext.toString());
```
please don't use system out anywhere

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/ValidateTest.java:
```diff
@@ -0,0 +1,46 @@
+        validState(true, EXPECTED_MESSAGE);
+
+        try {
+            validState(false, EXPECTED_MESSAGE);
```
assertThrows returns instance of the exception

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/ValidateTest.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Test
+    public void instantiationNotAllowed() {
+        Executable instantiationNotAllowed = () -> {
```
please inline it

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/WiremockFactoryTest.java:
```diff
@@ -0,0 +1,115 @@
+        when(mockedServer.factory()).thenReturn((Class) PrivateClassNotAllowed.class);
+
+        try {
+            factory.createServer(mockedServer);
```
please use assertThrows

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/WiremockFactoryTest.java:
```diff
@@ -0,0 +1,115 @@
+        when(mockedServer.customizer()).thenReturn((Class) PrivateClassNotAllowed.class);
+
+        try {
+            factory.createCustomizer(mockedServer);
```
here too

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/WiremockResolverUnitTest.java:
```diff
@@ -0,0 +1,135 @@
+ */
+public class WiremockResolverUnitTest {
+    private static final int EXPECTED_PORT = 8088;
+    private WiremockFactory wiremockFactory = mock(WiremockFactory.class);
```
please use mockito junit5 extension and annotaded fields or test arguments, now it looks like a mess :)

lanwen commented on src/test/java/ru/lanwen/wiremock/ext/WiremockResolverUnitTest.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    @BeforeEach
+    public void setup() throws Exception {
```
I think its too much efforts to do the test - integration test covers the same in a much better way

lanwen reviewed (changes requested):
tests should be simplified a bit

SourcePond committed:
Simplified tests
- Use assertThrows where applicable
- Inlined Executable where necessary
- Simplified WiremockResolverUnitTest, removed tests which are already covered by integration test

lanwen commented:
Many thanks for your changes :) will merge and release it during the day
